### PR TITLE
Clarify node affinity API

### LINKED ISSUE #31769: Node Affinity NodeSelectorTerms description is wrong
> If you specify multiple nodeSelectorTerms associated with nodeAffinity types, then the pod can be scheduled onto a node **if one of** the nodeSelectorTerms can be satisfied.

> If you specify multiple matchExpressions associated with nodeSelectorTerms, then the pod can be scheduled onto a node **only if all* matchExpressions is satisfied.

Should be, 

If you specify multiple matchExpressions associated with nodeSelectorTerms, then the pod can be scheduled onto a node **if one of** matchExpressions can be satisfied.

If you specify multiple node label requirements associated with matchExpressions types, then the pod can be scheduled onto a node **only if all** requirements can be satisfied.

### PR DESCRIPTION
**Pull request by alculquicondor**
Fixes #31769

I don't think there was anything wrong with the existing wording, but it was confusing. It was not clear whether it was referring to the mentions of the field or the items within the field.

Also replace @bsalamat with @alculquicondor, as a SIG Scheduling TL

### CODE REVIEW

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 1]
```diff
@@ -144,13 +144,12 @@ to repel Pods from specific nodes.
...
-satisfied.
+If you specify multiple terms in `nodeSelectorTerms` associated with `nodeAffinity`
+types, then the Pod can be scheduled onto a node if one of the specified terms
+can be satisfied (terms are ORed).
 
-If you specify multiple `matchExpressions` associated with a single `nodeSelectorTerms`,
-then the Pod can be scheduled onto a node only if all the `matchExpressions` are
-satisfied. 
+If you specify multiple expressions in `matchExpressions` associated with a single term in `nodeSelectorTerms`,
+then the Pod can be scheduled onto a node only if all the expressions are satisfied.
```
Comment (cortespao) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:152]:
```suggestion
If you specify multiple expressions in a single `matchExpressions` associated in a `nodeSelectorTerms`,
then the Pod can be scheduled onto a node only if _all_ the expressions are satisfied (expressions are ANDed). 
```

Comment (alculquicondor) [reply content/en/docs/concepts/scheduling-eviction/assign-pod-node.md]: added the word "field"

**Pull request merged**